src/serialbox-fortran/m_serialize.f90

```diff
@@ -296,6 +307,10 @@ END SUBROUTINE fs_disable_serialization
       fs_read_double_4d
   END INTERFACE
 
+#ifdef SERIALBOX_FORTRAN_SAVEPOINT_UNIQUE_ID
```

Comment:
What's this about? Doesn't sound like it should be a compile-time switch.

Reply:
This was a looooong discussion with Fabian about the different use-cases of serialbox2. Basically, there are two conceptual models of what serialization is:
- Database model: Every (savepoint, field) has to be unique.
- Sequental model: As the code is being run, it encounters a series of savepoints and data statements.
Serialbox2 was implemented as the former and imposes this uniqueness. This might be a good match for object oriented C++ programs, but it is not for sequential Fortran programs. In order to make every savpoint unique (say of a save point somewhere deep down in a subroutine), one has to expose global data (e.g. timestep, sub-cycling step, additional meta-info) available in this subroutine in order to declare a unique savepoint. Fabian did not let himself be convinced that this is not a good idea.
Now, in a real code such as FV3 this is an issue. FV3 uses significantly more subroutine call levels as compared to COSMO. So I have opted to implement a feature where each savepoint is attached a unique ID that is simply counted up, which makes each savepoint unique. Since this breaks with the current paradigm of serialbox2, I made this a Fortran only feature and retained it as an #ifdef.

Reply:
I would like to avoid adding compile time switches. I am thinking of installing packages with a package manager on a system. Do you really want to have 2 serialbox installations, one with this feature enabled and one disabled? We could solve it by passing a boolean to enable it to `fs_create_savepoint`.

Reply:
I went for option 2 in the above list. It is implemented in #222 .